### PR TITLE
Support for MUSL

### PR DESCRIPTION
**Pull request by WorksButNotTested**

### CODE REVIEW

File: gum/backend-linux/gumprocess-linux.c [hunk 1]
```diff
@@ -404,6 +404,7 @@ gum_process_modify_thread (GumThreadId thread_id,
   if (thread_id == gum_process_get_current_thread_id ())
   {
 #ifndef HAVE_ANDROID
+#ifndef HAVE_MUSLLIBC
```
Comment (oleavr) [inline gum/backend-linux/gumprocess-linux.c:407]:
```suggestion
#if !defined (HAVE_ANDROID) && !defined (HAVE_MUSLLIBC)
```

File: gum/backend-linux/gumprocess-linux.c [hunk 2]
```diff
@@ -421,6 +422,7 @@ gum_process_modify_thread (GumThreadId thread_id,
     }
 
     success = TRUE;
+#endif
```
Comment (oleavr) [inline gum/backend-linux/gumprocess-linux.c:425]:
```suggestion
```

File: gum/backend-linux/gumprocess-linux.c (hunk 1 above)
Comment (WorksButNotTested) [reply gum/backend-linux/gumprocess-linux.c]: Changed to:
#if !defined (HAVE_ANDROID) && (defined (__GLIC__) || defined (__UCLIBC__))

and accompanied by a comment.

File: gum/backend-linux/gumprocess-linux.c (hunk 2 above)
Comment (WorksButNotTested) [reply gum/backend-linux/gumprocess-linux.c]: Deleted

File: gum/backend-linux/gumprocess-linux.c [hunk 3]
```diff
@@ -403,7 +403,13 @@ gum_process_modify_thread (GumThreadId thread_id,
 
   if (thread_id == gum_process_get_current_thread_id ())
   {
-#ifndef HAVE_ANDROID
+/*
+ * getcontext/setcontext is not supported on the musl c-runtime (or android).
+ * musl, however doesn't provide any pre-processor definitions which allow it to
+ * be readily identified. However, the other major runtimes do, so we use their
+ * absence to determine that musl is in use and hence omit the block.
+ */
```
Comment (oleavr) [inline gum/backend-linux/gumprocess-linux.c:411]:
```suggestion
    /*
     * getcontext/setcontext is not supported on the musl C-runtime (or Android).
     * musl, however doesn't provide any pre-processor definitions which allow it to
     * be readily identified. However, the other major runtimes do, so we use their
     * absence to determine that musl is in use and hence omit the block.
     */
```

File: gum/backend-linux/gumprocess-linux.c [hunk 4]
```diff
@@ -403,7 +403,13 @@ gum_process_modify_thread (GumThreadId thread_id,
...
+#if !defined (HAVE_ANDROID) && (defined (__GLIC__) || defined (__UCLIBC__))
```
Comment (oleavr) [inline gum/backend-linux/gumprocess-linux.c:412]:
```suggestion
#if !defined (HAVE_ANDROID) && (defined (__GLIBC__) || defined (__UCLIBC__))
```

Typo for glibc, right?

File: gum/backend-linux/gumprocess-linux.c [hunk 5]
```diff
@@ -403,7 +403,13 @@ gum_process_modify_thread (GumThreadId thread_id,
 
   if (thread_id == gum_process_get_current_thread_id ())
   {
-#ifndef HAVE_ANDROID
+    /*
+     * getcontext/setcontext is not supported on the musl C-runtime (or Android).
+     * musl, however doesn't provide any pre-processor definitions which allow it to
+     * be readily identified. However, the other major runtimes do, so we use their
+     * absence to determine that musl is in use and hence omit the block.
+     */
```
Comment (oleavr) [inline gum/backend-linux/gumprocess-linux.c:411]: Needs a reflow to 80 cols max. Sorry I forgot to think of that :/

**Comment by WorksButNotTested** [comment]
Applied changes.

**Pull request merged**